src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/TestRulesContext.cs

```diff
@@ -51,25 +51,47 @@ public T CreateRule<T>() where T : ConsensusRuleBase, new()
         }
     }
 
+    public class RuleRegistrationHelper
+    {
+        public T RegisterRule<T>(ConsensusRuleEngine ruleEngine) where T : ConsensusRuleBase, new()
+        {
+            var rule = new T();
+
+            if (rule is IHeaderValidationConsensusRule validationConsensusRule)
+                ruleEngine.Network.Consensus.HeaderValidationRules = new List<IHeaderValidationConsensusRule>() { validationConsensusRule };
+
+            if (rule is IIntegrityValidationConsensusRule consensusRule)
+                ruleEngine.Network.Consensus.IntegrityValidationRules = new List<IIntegrityValidationConsensusRule>() { consensusRule };
+
+            if (rule is IPartialValidationConsensusRule partialValidationConsensusRule)
+                ruleEngine.Network.Consensus.PartialValidationRules = new List<IPartialValidationConsensusRule>() { partialValidationConsensusRule };
+
+            if (rule is IFullValidationConsensusRule fullValidationConsensusRule)
+                ruleEngine.Network.Consensus.FullValidationRules = new List<IFullValidationConsensusRule>() { fullValidationConsensusRule };
+
+            ruleEngine.Register();
```

Comment:
what happens if the rule is not one of those types?

Reply:
Shouldn't be possible because there are only 4 rule types.

Reply:
fixed